### PR TITLE
Add hooks for user interactions

### PR DESCRIPTION
**Pull request by robgrant**

### CODE REVIEW

File: Pod/Classes/PerspectivePhotoBrowserViewController.swift [hunk 1]
```diff
@@ -1,6 +1,12 @@
 
 import UIKit
 
+public protocol PerspectivePhotoBrowserViewControllerDelegate: class {
+  func perspectivePhotoBrowserViewController(_ perspectivePhotoBrowserViewController: PerspectivePhotoBrowserViewController, didViewPhotoAt index: Int)
+  func perspectivePhotoBrowserViewControllerDidZoomPhoto(_ perspectivePhotoBrowserViewController: PerspectivePhotoBrowserViewController)
+  func perspectivePhotoBrowserViewControllerDidFinish(_ perspectivePhotoBrowserViewController: PerspectivePhotoBrowserViewController)
```
Comment (robgrant) [inline Pod/Classes/PerspectivePhotoBrowserViewController.swift]: I originally implemented this with closures, but I think a delegate is more conventional and therefore more suitable for an open source library. However, I didn't consider that all of the methods would have to be implemented. I'll adjust.

File: Pod/Classes/PerspectivePhotoBrowserViewController.swift [hunk 2]
```diff
@@ -44,19 +55,20 @@ public class PerspectivePhotoBrowserViewController: UIViewController, Perspectiv
...
-  func swipedVertically() {
-    self.dismiss(animated: true, completion: .none)
-  }
-  
-  // MARK: Actions
-  @IBAction func userDidPress(doneButton sender: UIButton) {
-    self.dismiss(animated: true, completion: .none)
+  @IBAction func done() {
+    guard let delegate = delegate else {
+      self.dismiss(animated: true, completion: .none)
```
Comment (robgrant) [inline Pod/Classes/PerspectivePhotoBrowserViewController.swift]: I included the check for backward compatibility—without this, the consumer must conform to the delegate protocol to dismiss the photo browser.

File: Pod/Classes/PerspectivePhotoBrowserViewController.swift (hunk 1 above)
Comment (amol-c) [inline Pod/Classes/PerspectivePhotoBrowserViewController.swift]: sounds good.

File: Pod/Classes/PerspectivePhotoBrowserViewController.swift (hunk 2 above)
Comment (amol-c) [inline Pod/Classes/PerspectivePhotoBrowserViewController.swift]: So, you didn't want the dismiss to be handled here ? and want it to be handled by the `perspectivePhotoBrowserViewControllerDidFinish` ?

Is there a particular reason to do so ?

**Pull request merged**